### PR TITLE
UserRegister functionality and tests

### PR DESCRIPTION
**Pull request by linajin03**
Was able to create a create user functionality. The main issue is deleting the created users from the database from running tests that succesfully added new users into the database.

Added new classes and interfaces:
- UserRegister
- UserRegisterCreateUser 
- UserRegisterInputs
- UserRegisterReponseFormatter
- UserRegisterStatus
- UserRegistrationFailed
- UserRegisterTest

UserRegistrationFailed connects with the screen to display failed message (front-end please let me know if the format is okay!)

### CODE REVIEW

**Review by AlexChenl3 — DISMISSED** [review]
everthing looks good, i like the usernames u picked out for ur tests

File: src/main/java/UserRegister.java [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+        this.status = status;
+    }
+
+    /**
+     * Checks whether inputtedUsername exists in DataBase
+     * @return UserRegisterStatus status with status failure or success, depending on if the User was able to be created.
+     */
+    @Override
+    public UserRegisterInputs create(UserRegisterInputs inputs){
+        if (this.inputs.checkUserExists()){
```
Comment (Ljime) [inline src/main/java/UserRegister.java:18]: Docstring parameter missing here

File: src/main/java/UserRegistrationFailed.java [hunk 2]
```diff
@@ -0,0 +1,8 @@
+public class UserRegistrationFailed extends RuntimeException{
+    /**
+     * Constructs a new runtime exception with message errorMessage
+     */
+    public UserRegistrationFailed(String errorMessage){
+        super(errorMessage);
```
Comment (Ljime) [inline src/main/java/UserRegistrationFailed.java:6]: Here as well

File: src/main/java/UserRegisterResponseFormatter.java [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+    @Override
+    public UserRegisterInputs showSuccess(UserRegisterInputs newUser) {
+        return newUser;
+    }
+
+    /**
+     * Shows fail message by calling UserRegistrationFailed
+     */
+    @Override
+    public UserRegisterInputs showFailure(String failMessage) {
```
Comment (Ljime) [inline src/main/java/UserRegisterResponseFormatter.java:13]: Same here

**Review by Ljime — CHANGES_REQUESTED** [review]
Overall logic looks good, only thing is that some functions doc strings are missing the @param part of the docstring

**Review by chinchinjim — APPROVED** [review]
looks good, like the clean architecture structure! ditto on what hermanns said about the docstrings, you could also add docstrings to the testing functions but its not necessary since the function names are pretty descriptive already. nice work Lina! :)

**Review by Ljime — APPROVED** [review]

**Review by AlexChenl3 — APPROVED** [review]
reapproving

**Review by kevinle623 — APPROVED** [review]
great job implementing the user account logic, looks good to me

**Pull request merged**